scripts/nodes/setup-ros-node.bash

```diff
@@ -0,0 +1,340 @@
+#!/bin/bash
+#
+# Copyright 2021 Denis Stogl (Stogl Robotics Consulting)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+usage="setup-ros-node FILE_NAME [PKG_DEPS] [CLASS_NAME]"
+
+# Load Framework defines
+script_own_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null && pwd)"
+source $script_own_dir/../../setup.bash
+check_and_set_ros_distro_and_version "${ROS_DISTRO}"
+
+FILE_NAME=$1
+if [ -z "$FILE_NAME" ]; then
+  print_and_exit "ERROR: You should provide robot name! Nothing to do 😯" "$usage"
+fi
+
+PKG_DEPS=$2
+if [ -z "$PKG_DEPS" ]; then
+  PKG_DEPS=""
+fi
+
+if [ ! -f "package.xml" ]; then
+  print_and_exit "ERROR: 'package.xml' not found. You should execute this script at the top level of your package folder. Nothing to do 😯" "$usage"
+fi
+PKG_NAME="$(grep -Po '(?<=<name>).*?(?=</name>)' package.xml | sed -e 's/[[:space:]]//g')"
+
+CLASS_NAME=$3
+if [ -z "$CLASS_NAME" ]; then
+  delimiter='_'
+  s="$FILE_NAME$delimiter"
+  CLASS_NAME=""
+  while [[ $s ]]; do
+    part="${s%%"$delimiter"*}"
+    s=${s#*"$delimiter"}
+    CLASS_NAME="$CLASS_NAME${part^}"
+  done
+  echo -e "${TERMINAL_COLOR_USER_CONFIRMATION}ClassName guessed from the '$FILE_NAME': '$CLASS_NAME'. Is this correct? If not, provide it as the second parameter.${TERMINAL_COLOR_NC}"
```

Comment:
line 39, it says the `FILE_NAME` is first parameter, `PKG_DEPS` is second and `CLASS_NAME` is only third. Here it says provide it as the *second parameter*, which does not work.

If i only call 
`setup-ros-node my_node`, `PKG_DEPS` is `""`, which is expected

However, if i call
`setup-ros-node my_node MyFirstNode`, the class name is still being guessed.

Below, I removed the `PKG_DEPS` argument, as it is not used anywhere.

Reply:
you are right! I have fixed nos using PKG_DEPS. Can you check again?

The thing is that as we are using positional arguments, you have to provide empty second argument to set the third.

Reply:
then, if i want to define a special class name, I have to know to that the second parameter should be empty, which seems convoluted.

For example, now running:
```
setup-ros-node my_node MyFirstNode
```
places `<depend>MyFirstNode</depend>` in the `package.xml`.


In my opinion, this script is trying to do too much.

This script should generate the required files and boilerplate for `CMakeLists.txt` and ROS parameters.
Adding dependancies should be done by the user, manually, as we usually add dependancies anyways while developing the node.

I suggest we remove adding package dependancies except `<build_depend>generate_parameter_library</build_depend>`, and make the custom node name parameter `$2`, in the remote case the class name will not correspond to the file name, but CamelCase.

Reply:
In the spirit of my previous comment, can we remove the `is the package already configured?` prompt?

when using `create-new-package`, we are already "configuring the package" by setting up some `CMakeLists.txt` [code](https://github.com/StoglRobotics/ros_team_workspace/blob/b289773332ee4956c66faac11ad91418c5b719ab/scripts/create-new-package.bash#L184-L220).

We can just make this script add dependancy to `generate_parameter_library`, link the new files and call it a day? This prompt also adds some unecessary complexity, in my opinion.

We can always expand the functionality later, as we use the script.

Reply:
the functionality about adding the `is already configured` is when you are adding a second node to already setup package. Not wehn you configure your package with `create-new-package`.

Reply:
On the other hand, I am lazy, and would rather do small fixes later for the dependecies then have to add them manually. We are talking here not about 100% dependecy coverage when you gnerate your file, but more about initial dependecies.